### YamlMime:ManagedReference
items:
  - uid: com.microsoft.azure.elasticdb.query.exception
    id: exception
    children:
      - com.microsoft.azure.elasticdb.query.exception._multi_shard_aggregate_exception
      - com.microsoft.azure.elasticdb.query.exception._multi_shard_exception
      - com.microsoft.azure.elasticdb.query.exception._multi_shard_partial_read_exception
      - com.microsoft.azure.elasticdb.query.exception._multi_shard_result_set_closed_exception
      - com.microsoft.azure.elasticdb.query.exception._multi_shard_result_set_internal_exception
      - com.microsoft.azure.elasticdb.query.exception._multi_shard_schema_mismatch_exception
    href: com.microsoft.azure.elasticdb.query.exception.yml
    langs:
      - java
    name: com.microsoft.azure.elasticdb.query.exception
    nameWithType: com.microsoft.azure.elasticdb.query.exception
    fullName: com.microsoft.azure.elasticdb.query.exception
    type: Namespace
    source:
      remote:
        path: elastic-db-tools/src/main/java/com/microsoft/azure/elasticdb/query/exception/MultiShardAggregateException.java
        branch: master
        repo: 'https://github.com/Microsoft/elastic-db-tools-for-java'
      path: elastic-db-tools/src/main/java/com/microsoft/azure/elasticdb/query/exception/MultiShardAggregateException.java
    syntax:
      content: package com.microsoft.azure.elasticdb.query.exception
references:
  - uid: com.microsoft.azure.elasticdb.query.exception._multi_shard_aggregate_exception
    parent: com.microsoft.azure.elasticdb.query.exception
    href: com.microsoft.azure.elasticdb.query.exception._multi_shard_aggregate_exception.yml
    name: MultiShardAggregateException
    nameWithType: MultiShardAggregateException
    fullName: com.microsoft.azure.elasticdb.query.exception.MultiShardAggregateException
    type: Class
    summary: '<p><span data-ttu-id="a4254-101">シャード セット間でクエリを実行するときに発生した 1 つまたは複数の例外エラーを表します。</span><span class="sxs-lookup"><span data-stu-id="a4254-101">Represents one or more Exception errors that occurred when executing a query across a shard set.</span></span> <span data-ttu-id="a4254-102">InnerExceptions フィールドがこれらの例外を収集し、さらに検査または処理のため InnerExceptions を反復処理できる 1 つ。</span><span class="sxs-lookup"><span data-stu-id="a4254-102">The InnerExceptions field collects these exceptions and one can iterate through the InnerExceptions for further inspection or processing.</span></span> <span data-ttu-id="a4254-103">複数のシャード間で発生したエラーの情報をやり取りする目的: パブリック型</span><span class="sxs-lookup"><span data-stu-id="a4254-103">Purpose: Public type that communicates errors that occurred across multiple shards</span></span> </p>'
    syntax:
      content: public class MultiShardAggregateException
  - uid: com.microsoft.azure.elasticdb.query.exception._multi_shard_exception
    parent: com.microsoft.azure.elasticdb.query.exception
    href: com.microsoft.azure.elasticdb.query.exception._multi_shard_exception.yml
    name: MultiShardException
    nameWithType: MultiShardException
    fullName: com.microsoft.azure.elasticdb.query.exception.MultiShardException
    type: Class
    summary: '<p><span data-ttu-id="a4254-104">DEVNOTE:、シャード ShardLocation のシリアル化のシリアル化可能なサポートのないので、ここで型が、正常に動作では SMM ShardLocation がカプセル化します。</span><span class="sxs-lookup"><span data-stu-id="a4254-104">DEVNOTE: Encapsulate SMM ShardLocation type for now since Shard isn''t Serializable Support for serialization of ShardLocation is in the works.</span></span> <span data-ttu-id="a4254-105">A <xref uid="com.microsoft.azure.elasticdb.query.exception._multi_shard_exception" data-throw-if-not-resolved="false">MultiShardException</xref>シャードに対して操作を実行するときに発生した例外を表します。</span><span class="sxs-lookup"><span data-stu-id="a4254-105">A <xref uid="com.microsoft.azure.elasticdb.query.exception._multi_shard_exception" data-throw-if-not-resolved="false">MultiShardException</xref> represents an exception that occurred when performing operations against a shard.</span></span> <span data-ttu-id="a4254-106">シャードと発生した例外の両方の id に関する情報を提供します。</span><span class="sxs-lookup"><span data-stu-id="a4254-106">It provides information about both the identity of the shard and the exception that occurred.</span></span> <span data-ttu-id="a4254-107">、例外の性質によっては 1 つことができます、複数のシャードのクエリを再実行してくださいし、直接に、例外を生成するシャードを対象としたクエリを実行するまたは最後に SSMS などの一般的なツールを使用して、シャードに対して手動でクエリを実行します。</span><span class="sxs-lookup"><span data-stu-id="a4254-107">Depending on the nature of the exception, one can try re-running the multi-shard query, execute a separate query targeted directly at the shard(s) on that yielded the exception, or lastly execute the query manually against the shard using a common tool such as SSMS.</span></span> </p>'
    syntax:
      content: public class MultiShardException
  - uid: com.microsoft.azure.elasticdb.query.exception._multi_shard_partial_read_exception
    parent: com.microsoft.azure.elasticdb.query.exception
    href: com.microsoft.azure.elasticdb.query.exception._multi_shard_partial_read_exception.yml
    name: MultiShardPartialReadException
    nameWithType: MultiShardPartialReadException
    fullName: com.microsoft.azure.elasticdb.query.exception.MultiShardPartialReadException
    type: Class
    summary: '<p><span data-ttu-id="a4254-108">MultiShardResultSet は、基になるシャードのいずれかからデータを読み取る例外にヒットしたときに、この例外をスローします。</span><span class="sxs-lookup"><span data-stu-id="a4254-108">The MultiShardResultSet throws this exception when an exception has been hit reading data from one of the underlying shards.</span></span> <span data-ttu-id="a4254-109">これは、すべての行が対象のシャードから正常に取得されたことを示します。</span><span class="sxs-lookup"><span data-stu-id="a4254-109">This indicates that not all rows have been successfully retrieved from the targeted shard(s).</span></span> <span data-ttu-id="a4254-110">ユーザー、手順を実行、クエリを再実行するかどうか、または既に取得されている行を含む作業を続行するかどうかを判断するために必要です。</span><span class="sxs-lookup"><span data-stu-id="a4254-110">Users can then take the steps necessary to decide whether to re-run the query, or whether to continue working with the rows that have already been retrieved.</span></span> <span data-ttu-id="a4254-111">結果の一部のポリシーでは、この例外がスローされますのみです。</span><span class="sxs-lookup"><span data-stu-id="a4254-111">This exception is only thrown with the partial results policy.</span></span> <span data-ttu-id="a4254-112">目的: MultiShardResultSet 基になるの結果セットのいずれかの next() 呼び出し中に例外が発生するとをスローする例外をカスタムです。</span><span class="sxs-lookup"><span data-stu-id="a4254-112">Purpose: Custom exception to throw when the MultiShardResultSet hits an exception during a next() call to one of the underlying ResultSets.</span></span> <span data-ttu-id="a4254-113">そのためことでしたそのシャードからすべての結果を読み込むことがわかってすべてが発生したときに、何らかの理由でユーザーに通知する必要があります。</span><span class="sxs-lookup"><span data-stu-id="a4254-113">When that happens all we know is that we were not able to read all the results from that shard, so we need to notify the user somehow.</span></span> </p>'
    syntax:
      content: public class MultiShardPartialReadException extends MultiShardException
  - uid: com.microsoft.azure.elasticdb.query.exception._multi_shard_result_set_closed_exception
    parent: com.microsoft.azure.elasticdb.query.exception
    href: com.microsoft.azure.elasticdb.query.exception._multi_shard_result_set_closed_exception.yml
    name: MultiShardResultSetClosedException
    nameWithType: MultiShardResultSetClosedException
    fullName: com.microsoft.azure.elasticdb.query.exception.MultiShardResultSetClosedException
    type: Class
    summary: '<p><span data-ttu-id="a4254-114">独自の例外をスロー、MultiShardResultSet が閉じられたときと、ユーザーが閉じたリーダーに対して操作を実行しようとするとします。</span><span class="sxs-lookup"><span data-stu-id="a4254-114">Custom exception to throw when the MultiShardResultSet is closed and the user attempts to perform an operation on the closed reader.</span></span> <span data-ttu-id="a4254-115">目的: throw、MultiShardResultSet が閉じられたときと、ユーザーにカスタム例外は、いくつかの操作を試行します。</span><span class="sxs-lookup"><span data-stu-id="a4254-115">Purpose: Custom exception to throw when the MultiShardResultSet is closed and the user attempts to perform some operation.</span></span> </p>'
    syntax:
      content: public class MultiShardResultSetClosedException
  - uid: com.microsoft.azure.elasticdb.query.exception._multi_shard_result_set_internal_exception
    parent: com.microsoft.azure.elasticdb.query.exception
    href: com.microsoft.azure.elasticdb.query.exception._multi_shard_result_set_internal_exception.yml
    name: MultiShardResultSetInternalException
    nameWithType: MultiShardResultSetInternalException
    fullName: com.microsoft.azure.elasticdb.query.exception.MultiShardResultSetInternalException
    type: Class
    summary: '<p><span data-ttu-id="a4254-116">MultiShardResultSet が無効な状態のときにスローされるカスタムの例外。</span><span class="sxs-lookup"><span data-stu-id="a4254-116">Custom exception that is thrown when the MultiShardResultSet is in an invalid state.</span></span> <span data-ttu-id="a4254-117">この例外を繰り返し発生する場合、マイクロソフト カスタマー サポートにお問い合わせください。</span><span class="sxs-lookup"><span data-stu-id="a4254-117">If you experience this exception repeatedly, please contact Microsoft Customer Support.</span></span> <span data-ttu-id="a4254-118">目的: MultiShardResultSet が無効な状態のときにスローする例外をカスタムです。</span><span class="sxs-lookup"><span data-stu-id="a4254-118">Purpose: Custom exception to throw when the MultiShardResultSet is in an invalid state.</span></span> <span data-ttu-id="a4254-119">このエラーをしないように、ユーザー。</span><span class="sxs-lookup"><span data-stu-id="a4254-119">This error should not make it out to the user.</span></span> </p>'
    syntax:
      content: public class MultiShardResultSetInternalException
  - uid: com.microsoft.azure.elasticdb.query.exception._multi_shard_schema_mismatch_exception
    parent: com.microsoft.azure.elasticdb.query.exception
    href: com.microsoft.azure.elasticdb.query.exception._multi_shard_schema_mismatch_exception.yml
    name: MultiShardSchemaMismatchException
    nameWithType: MultiShardSchemaMismatchException
    fullName: com.microsoft.azure.elasticdb.query.exception.MultiShardSchemaMismatchException
    type: Class
    summary: '<p><span data-ttu-id="a4254-120">カスタム例外は、全体的なクエリに参加しているシャードの少なくとも 1 つのスキーマが、複数のシャード クエリ全体の予期されるスキーマに準拠していない場合にスローされます。</span><span class="sxs-lookup"><span data-stu-id="a4254-120">Custom exception thrown when the schema on at least one of the shards participating in the overall query does not conform to the expected schema for the multi-shard query as a whole.</span></span> <span data-ttu-id="a4254-121">目的: 特定のシャードから結果セットからスキーマが全体としてファンアウト クエリに対して予期されるスキーマに準拠していない場合にスローされる例外をカスタムです。</span><span class="sxs-lookup"><span data-stu-id="a4254-121">Purpose: Custom exception to throw when the schema from a ResultSet from a given shard does not conform to the expected schema for the fanout query as a whole.</span></span> </p>'
    syntax:
      content: public class MultiShardSchemaMismatchException extends MultiShardException